docs/source/optimization_ov.mdx

```diff
@@ -112,6 +112,81 @@ metrics = trainer.evaluate()
 trainer.save_model()
 ```
 
+## Distillation
+
+Distillation is an effective method to improve model performance by learning from the knowledge of, usually, larger model. `OVTrainer` also provides the capability to perform distillation during training. Users are required to load the teacher and feed the loaded model to `OVTrainer` as follows. Subclassed from `TrainingArguments`, `OVTrainingArguments` is provided to ease users to specify distillation hyperparameters (such as teacher model path, distillation weightage and temperature), as well as to expose the hyperparameters as commandline arguments to the training script.
+
+```diff
+-from optimum.intel.openvino.trainer import OVConfig, OVTrainer
++from optimum.intel.openvino import OVConfig, OVTrainer, OVTrainingArguments
+
+# Load teacher model
++teacher_model = AutoModelForSequenceClassification.from_pretrained(teacher_model_or_path)
+
+trainer = OVTrainer(
+    model=model,
++   teacher_model=teacher_model,
+-    args=TrainingArguments(save_dir, num_train_epochs=1.0, do_train=True, do_eval=True),
++    args=OVTrainingArguments(save_dir, num_train_epochs=1.0, do_train=True, do_eval=True, distillation_temperature=3, distillation_weight=0.9),
+    train_dataset=dataset["train"].select(range(300)),
+    eval_dataset=dataset["validation"],
+    compute_metrics=compute_metrics,
+    tokenizer=tokenizer,
+    data_collator=default_data_collator,
+    ov_config=ov_config,
+    feature="sequence-classification",
+)
+```
+
+## Advanced Compression with NNCF
+
+`OVTrainer` integrates NNCF as backend of model optimization. It exposes most of the model compression algorithms in NNCF, variants of quantization, pruning to name a few. Users only need to provide a config dictionary to apply the selected NNCF algorithm(s). For example, NNCF provides structured pruning capability which closely resembles the work of [Lagunas et al., 2021, Block Pruning For Faster Transformers](https://arxiv.org/pdf/2109.04838.pdf), following is the config dictionary to apply the algorithm.
+
+```python
+compression_config = {
+    "compression":
+    {
+    "algorithm":  "movement_sparsity",
+    "params": {
+        "warmup_start_epoch":  1,
+        "warmup_end_epoch":    4,
+        "importance_regularization_factor":  0.01,
+        "enable_structured_masking":  True
+    },
+    "sparse_structure_by_scopes": [
+        {"mode":  "block",   "sparse_factors": [32, 32], "target_scopes": "{re}.*BertAttention.*"},
+        {"mode":  "per_dim", "axis":  0,                 "target_scopes": "{re}.*BertIntermediate.*"},
+        {"mode":  "per_dim", "axis":  1,                 "target_scopes": "{re}.*BertOutput.*"},
+    ],
+    "ignored_scopes": ["{re}.*NNCFEmbedding", "{re}.*qa_outputs*", "{re}.*LayerNorm.*"]
```

Comment:
In this example, the task is text classification so I would remove the part of the configuration adapted for a QA task

Reply:
Good catch, patched for text classification in https://github.com/huggingface/optimum-intel/pull/150/commits/e93045ec3bf6058943a2f29ee7f9d7a48d35a451